LabDrivers/utils.py

```diff
@@ -254,6 +265,135 @@ def is_IP_port(device_port, return_vals=False):
     else:
         return True
 
+def list_properties():
+    """
+        This returns a dictionary of all drivers that have properties dictionary in form:
+        return {
+            'NAME': object
+        }
+    """
+    interface = [INTF_VISA, INTF_PROLOGIX, INTF_SERIAL]
+
+    #    instruments.append('TIME')
+    properties = {}
+    #    params['TIME']=[]
+
+    # list all the .py files in the drivers folder
+    my_path = inspect.getfile(list_drivers).rstrip('utils.py')
+    driver_files = os.listdir(my_path)
+
+    for file_name in driver_files:
+
+        if file_name.endswith('.py') \
+                and (
+                not file_name == 'Tool.py' and not file_name == '__init__.py' and not file_name == 'utils.py'):  # yikes
+            name = file_name.split('.py')[0]
+
+            # import the module of the instrument in the package drivers
+            driver = import_module('.' + name, package=LABDRIVER_PACKAGE_NAME)
+
+            try:
+                driver_interface = driver.INTERFACE
+
+            except AttributeError:
+                logging.error(
+                    "The following module is probably not an instrument driver, "
+                    "please remove it from the package %s" % LABDRIVER_PACKAGE_NAME
+                )
+                driver_interface = ''
+
+            if hasattr(driver, 'properties'):
+                properties[name] = driver.properties
+
+    return properties
+
+def list_functions():
+    """
+        This returns a dictionary of all drivers that have function dictionary in form:
+        return {
+            'NAME': object
+        }
+    """
+    interface = [INTF_VISA, INTF_PROLOGIX, INTF_SERIAL]
+
+    #    instruments.append('TIME')
+    functions = {}
+    #    params['TIME']=[]
+
+    # list all the .py files in the drivers folder
+    my_path = inspect.getfile(list_drivers).rstrip('utils.py')
+    driver_files = os.listdir(my_path)
+
+    for file_name in driver_files:
+
+        if file_name.endswith('.py') \
+                and (
+                not file_name == 'Tool.py' and not file_name == '__init__.py' and not file_name == 'utils.py'):  # yikes
+            name = file_name.split('.py')[0]
+
+            # import the module of the instrument in the package drivers
+            driver = import_module('.' + name, package=LABDRIVER_PACKAGE_NAME)
+
+            try:
+                driver_interface = driver.INTERFACE
+
+            except AttributeError:
+                logging.error(
+                    "The following module is probably not an instrument driver, "
+                    "please remove it from the package %s" % LABDRIVER_PACKAGE_NAME
+                )
+                driver_interface = ''
+
+            if hasattr(driver, 'functions'):
+                functions[name] = driver.functions
+
+    return functions
+
+def list_plot_axes():
```

Comment:
```suggestion
def list_driver_plot_axes():
```

Reply:
same as above comment:

"This would be a good idea. The sole reason I separated all of these functions in the first place was to avoid passing extraneous information (which would then require further processing in FunctionWidget and PropertiesWidget) in list_drivers(). Done via PyCharm refactor"

worth noting that its sole call appeared to be in FunctionWidget